websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,285 @@
+			} else {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+				presenceData.buttons = [
+					{
+						label: "Read Article",
+						url: location.href
+					}
+				];
+			}
+			break;
+		}
+		case "forums.sonicretro.org": {
+			presenceData.smallImageKey = "forums_512";
+			if (search) {
+				const forumTitle = document.title.split(
+					" | Sonic and Sega Retro Forums"
+				)[0];
+				if (search.includes("?watched/forums")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?watched/threads")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched threads";
+				} else if (
+					search.startsWith("?find-new/") &&
+					search.includes("/posts")
+				) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing new posts";
+				} else if (
+					search.startsWith("?find-new/") &&
+					search.includes("/profile-posts")
+				) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing new profile posts";
+				} else if (search.startsWith("?forums/")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing a forum:";
+					presenceData.state = forumTitle;
+					presenceData.buttons = [
+						{
+							label: "View Forum",
+							url: location.href
+						}
+					];
+				} else if (search.startsWith("?threads/")) {
+					presenceData.largeImageKey = "reading";
+					presenceData.details = "Reading a thread:";
+					presenceData.state = forumTitle;
+					presenceData.buttons = [
+						{
+							label: "Read Thread",
+							url: location.href
+						}
+					];
+				} else if (search.startsWith("?members/")) {
+					if (document.title.startsWith("Notable Members")) {
+						presenceData.largeImageKey = "search";
+						presenceData.details = "Browsing notable members";
+					} else {
+						presenceData.largeImageKey = "reading";
+						presenceData.details = "Viewing a member's page";
+						presenceData.state = `@${forumTitle}`;
+						presenceData.buttons = [
+							{
+								label: "View Member",
+								url: location.href
+							}
+						];
+					}
+				} else if (search.startsWith("?search/")) {
+					presenceData.largeImageKey = "search";
+					if (
+						document.title.startsWith("Error") ||
+						document.title.startsWith("Search")
+					)
+						presenceData.details = "Searching for something...";
+					else {
+						presenceData.details = `Searching for ${
+							document.title
+								.split("Search Results for Query: ")[1]
+								.split(" | Sonic and Sega Retro Forums")[0]
+						}`;
+					}
+				} else if (
+					/(\?conversations\/|\?account\/|\?help\/)/.test(
+						document.location.search
+					) ||
+					location.href ===
+						"https://forums.sonicretro.org/index.php?forums/-/index.rss"
```

Comment:
I **highly** discourage you to compare against a URL this big, you should either check sth in the page or parts of the URL because this is very volatile